minigrid/__init__.py

```diff
@@ -565,6 +565,25 @@ def register_minigrid_envs():
         entry_point="minigrid.envs:UnlockPickupEnv",
     )
 
+    # WaveFunctionCollapse
+    # ----------------------------------------
+    try:
+        import imageio
```

Comment:
We could do this through looping through an pre-determined list of patterns (with a test to ensure that the all expected environments exist)

Reply:
My thinking was that maybe they shouldn't be registered if the dependencies are missing because they will show up as failing test cases to developers who are running tests without installing the optional dependencies. Is it easy to change the gymnasium testing infrastructure to skip those test in those situations?

Reply:
Yes, it should be possible to do that. 
My thinking is that the users should be able to test to create an environment and only at that time will an error be raised rather than the environment id not showing up the registry for an unknown reason at least to the user